vivarium/core/serialize.py

```diff
@@ -185,99 +181,76 @@ def deserialize(  # pylint: disable=arguments-differ
-    class Codec(TypeEncoder):
-        python_type = np.int64
-        def transform_python(self, value: np.int64) -> int:
-            return int(value)
-
-    def get_codecs(self) -> List:
-        return [self.Codec()]
-
-class NumpyInt32Serializer(Serializer):
-    """Serializer for ``np.int32`` objects."""
-    class Codec(TypeEncoder):
-        python_type = np.int32
-        def transform_python(self, value: np.int32) -> int:
-            return int(value)
-
-    def get_codecs(self) -> List:
-        return [self.Codec()]
-
-class NumpyFloat32Serializer(Serializer):
-    """Serializer for ``np.float32`` objects."""
-    class Codec(TypeEncoder):
-        python_type = np.float32
-        def transform_python(self, value: np.float32) -> float:
-            return float(value)
-
-    def get_codecs(self) -> List:
-        return [self.Codec()]
+    def serialize(self, data: Any) -> Union[List[str], str]:
+        try:
+            return_value = []
+            for subvalue in data:
+                return_value.append(f"!units[{str(subvalue)}]")
+            return return_value
+        except TypeError:
+            return f"!units[{str(data)}]"
 
 class SetSerializer(Serializer):
     """Serializer for set objects."""
-    class Codec(TypeEncoder):
-        python_type = set
-        def transform_python(self, value: set) -> List:
-            return list(value)
+    python_type = set
 
-    def get_codecs(self) -> List:
-        return [self.Codec()]
+    def serialize(self, data: set) -> List:
+        return list(data)
 
 class FunctionSerializer(Serializer):
     """Serializer for function objects."""
-    class Codec(TypeEncoder):
-        python_type = type(deserialize_value)
-        def transform_python(self, value: Callable) -> str:
-            return f"!FunctionSerializer[{str(value)}]"
+    python_type = type(deserialize_value)
+
+    def serialize(self, data: Callable) -> str:
+        return f"!FunctionSerializer[{str(data)}]"
 
-    def get_codecs(self) -> List:
-        return [self.Codec()]
 
 class ProcessSerializer(Serializer):
     """Serializer for processes if ``emit_process`` is enabled."""
-
-    def __init__(self) -> None:
-        super().__init__(name='processes')
+    python_type = Process
 
     def serialize(self, data: Process) -> str:
         proc_str = str(dict(data.parameters, _name=data.name))
         return f"!ProcessSerializer[{proc_str}]"
 
-# Subclasses of data types handled by custom
-# TypeEncoders require their own TypeEncoders.
-# This includes Process, Composites, etc.
-def get_codec_options() -> CodecOptions:
-    """Returns a set of options used for serializing and
-    deserializing BSON by collecting codecs from all registered
-    serializers."""
-    codecs = []
-    for serializer in serializer_registry.registry.values():
-        codecs += serializer.get_codecs()
-    return CodecOptions(type_registry=TypeRegistry(codecs))
+
+def make_default() -> Callable:
```

Comment:
can you name this something more descriptive? `make_default` is really not clear. Maybe `make_serializer_function`?

Reply:
Good idea. I’ll make that change as soon as I can.